src/operators.jl

```diff
@@ -858,3 +855,28 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+# special case for digraphs
+@traitfn function merge_vertices!(g::::IsDirected, v::Vector{U})  where U <: Integer
+    v0 = minimum(v)
+    vs = Set(v)
+    filter!(x -> x != v0,vs)
+    inedges = Set{eltype(g)}()
+    outedges = Set{eltype(g)}()
+    new_vertex_ids = collect(vertices(g))
```

Comment:
This becomes O(V) in memory. Is there a way to do this without collect? That is, if we assume that `length(vs) << nv(g)`, can we just keep track of the changes instead of keeping track of all the vertices?

We know that`rem_vertices!` returns a vertex map. Why do we need to create another one here?